### PR TITLE
Include readmes from examples into dashboard pages on build

### PR DESCRIPTION
**Pull request by g4s8**
#348 - First PR for including example readme files into dashboard help.

The build process:
 1. Copying `README.md` from examples to temporary build directory and rename it (on `generate-resources`)
 2. Convert markdown readmes to HTML pages (on `generate-resources`)
 3. Copy generated pages to classpath resources (on `process-resources`)

Dashboard repo page includes this readme as a part of `body` content if repo type matches expected readme type.

### CODE REVIEW

File: examples/binary/README.md [hunk 1]
```diff
@@ -13,8 +13,14 @@ repo:
...
-e.g. `http://localhost:8080/repo/libsqlite3.so`.
+Use `PUT` HTTP request to upload a file, and `GET` for downloading.</p>
+
+Example using [HTTPie](https://httpie.org/) CLI tool for uploading and downloading:
+<code>
+http -a {{user}}:password PUT https://central.artipie.com/{{user}}/{{name}}/file.bin @file.bin
+http GET https://central.artipie.com/{{user}}/{{name}}/file.bin --output=./file.bin
+</code>
+
+// TODO: add curl example
```
Comment (olenagerasimova) [inline examples/binary/README.md:26]: @g4s8 hm, markdown looks terrible now
image
I think we should not leave it like this

File: examples/docker/README.md [hunk 2]
```diff
@@ -12,22 +12,16 @@ repo:
...
+Docker registry has to be protected by HTTPS.
 
-To pull the image use the following command:
+Tag your image with `central.artipie.com/{{user}}/{{name}}` image prefix,
+and push it to central.artipie.com then. E.g.
+for `alpine:3.11` use:
+<pre>
+docker tag alpine:3.11 central.artipie.com/{{user}}/{{name}}/alpine:3.11
+docker push central.artipie.com/{{user}}/{{name}}/alpine:3.11
+</pre>
```
Comment (olenagerasimova) [inline examples/docker/README.md:23]: @g4s8 same here, could you use correct readme formatting please?

**Comment by g4s8** [comment]
@olenagerasimova thanks, should be fixed now.

**Comment by g4s8** [comment]
> Agree with @olenagerasimova, readme file with an example of usage would be turned into a mess after this change. Also, a purpose for which they were created will be completely lost.

@Sammers21 why? I almost didn't change the reames, only removed obsolete information (e.g. Docker doesn't require reverse proxy now) and added some usage examples.

**Comment by Sammers21** [comment]
@g4s8, the placeholders in the readme will make users angry. They came for an example, what they will get is far different:

image

**Comment by g4s8** [comment]
> @g4s8, the placeholders in the readme will make users angry. They came for an example, what they will get is far different:
> 
> image

@Sammers21 sounds reasonable. WDYT about using `http://localhost/artipie/` in examples and replace it with placeholder `https://central.artipie.com/{{user}}/{{repo}}` in compile time?

**Comment by 0crat** [comment]
@carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by g4s8** [comment]
Needs another solution here.

**Comment by 0crat** [comment]
Pull request [#354](https://github.com/artipie/artipie/issues/354) was not merged, no payment for `ARC`, see [§28](http://www.zerocracy.com/policy.html#28)

**Comment by 0crat** [comment]
Job audit: performer @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda)[/z](https://www.0crat.com/u/carlosmiranda) didn't make CR comments

**Comment by 0crat** [comment]
The project doesn't have enough funds, can't make a payment

**Comment by 0crat** [comment]
Code review was too long (27 days), architects (@g4s8, @olenagerasimova[/z](https://www.0crat.com/u/olenagerasimova)) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)

**Pull request closed without merging**